File: main-service/src/main/java/ru/pgk/main_service/common/mapper/Mappable.java
package ru.pgk.main_service.common.mapper;

import java.util.Collection;
import java.util.List;

public interface Mappable<E, D> {

    D toDto(E entity);

    List<D> toDto(Collection<E> entity);

    E toEntity(D dto);

    Collection<E> toEntity(Collection<D> dtos);

}

File: teacher-service/src/main/kotlin/ru/pgk/teacher_service/common/mapper/Mappable.kt
package ru.pgk.teacher_service.common.mapper

interface Mappable<E, D> {
    fun toDto(entity: E): D
    fun toDto(entity: Collection<E>): List<D>
    fun toEntity(dto: D): E
    fun toEntity(dto: Collection<D>): List<E>
}